scripts/lib/CIME/XML/expected_fails_file.py

```diff
@@ -0,0 +1,65 @@
+"""
+Interface to an expected failure xml file
+
+Here is an example:
+
+<?xml version= "1.0"?>
+<expectedFails version="1.0">
+  <test name="ERP_D_Ld10_P36x2.f10_f10_musgs.IHistClm50BgcCrop.cheyenne_intel.clm-ciso_decStart">
+    <phase name="RUN">
+      <status>FAIL</status>
+      <issue>#404</issue>
+    </phase>
+    <phase name="COMPARE_base_rest">
+      <status>PEND</status>
+      <issue>#404</issue>
+      <comment>Because of the RUN failure, this phase is listed as PEND</comment>
+    </phase>
+  </test>
+  <test name="PFS_Ld20.f09_g17.I2000Clm50BgcCrop.cheyenne_intel">
+    <phase name="GENERATE">
+      <status>FAIL</status>
+      <issue>ESMCI/cime#2917</issue>
+    </phase>
+    <phase name="BASELINE">
+      <status>FAIL</status>
+      <issue>ESMCI/cime#2917</issue>
+    </phase>
+  </test>
+</expectedFails>
+"""
+
+from CIME.XML.standard_module_setup import *
+
+from CIME.XML.generic_xml import GenericXML
+from CIME.XML.files import Files
+from CIME.expected_fails import ExpectedFails
+
+logger = logging.getLogger(__name__)
+
+class ExpectedFailsFile(GenericXML):
+
+    def __init__(self, infile, files=None):
+        if files is None:
+            files = Files()
+        # FIXME(wjs, 2019-01-02) Get schema, replacing the following line
```

Comment:
I still plan to add an xml schema definition. I'd like to get approval of my proposed schema, as per the example at the top of this file, before I write this xsd file.

Reply:
(This is why I have labeled this PR as WIP.)